src/platform/qpg/BLEManagerImpl.cpp

```diff
@@ -458,31 +462,31 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
                   "Advertisement data buffer is not big enough");
 
     // Fill in advertising data
-    index                = 0;
-    mAdvDataBuf[index++] = 0x02;                     // length
-    mAdvDataBuf[index++] = CHIP_ADV_DATA_TYPE_FLAGS; // AD type : flags
-    mAdvDataBuf[index++] = CHIP_ADV_DATA_FLAGS;      // AD value
-
-    mAdvDataBuf[index++] = static_cast<uint8_t>(mDeviceIdInfoLength + CHIP_ADV_SHORT_UUID_LEN + 1); // AD length
-    mAdvDataBuf[index++] = CHIP_ADV_DATA_TYPE_SERVICE_DATA;                                         // AD type : Service Data
-    mAdvDataBuf[index++] = chipUUID_CHIPoBLE_Service[1];                                            // AD value
-    mAdvDataBuf[index++] = chipUUID_CHIPoBLE_Service[0];
-    memcpy(&mAdvDataBuf[index], (void *) &mDeviceIdInfo, mDeviceIdInfoLength); // AD value
+    index               = 0;
+    advDataBuf[index++] = 0x02;                     // length
+    advDataBuf[index++] = CHIP_ADV_DATA_TYPE_FLAGS; // AD type : flags
+    advDataBuf[index++] = CHIP_ADV_DATA_FLAGS;      // AD value
+
+    advDataBuf[index++] = static_cast<uint8_t>(mDeviceIdInfoLength + CHIP_ADV_SHORT_UUID_LEN + 1); // AD length
+    advDataBuf[index++] = CHIP_ADV_DATA_TYPE_SERVICE_DATA;                                         // AD type : Service Data
+    advDataBuf[index++] = chipUUID_CHIPoBLE_Service[1];                                            // AD value
+    advDataBuf[index++] = chipUUID_CHIPoBLE_Service[0];
+    memcpy(&advDataBuf[index], (void *) &mDeviceIdInfo, mDeviceIdInfoLength); // AD value
     index = static_cast<uint8_t>(index + mDeviceIdInfoLength);
 
-    mAdvDataBuf[index++] = static_cast<uint8_t>(mDeviceNameLength + 1); // length
-    mAdvDataBuf[index++] = CHIP_ADV_DATA_TYPE_NAME;                     // AD type : name
-    memcpy(&mAdvDataBuf[index], mDeviceName, mDeviceNameLength);        // AD value
+    advDataBuf[index++] = static_cast<uint8_t>(mDeviceNameLength + 1); // length
+    advDataBuf[index++] = CHIP_ADV_DATA_TYPE_NAME;                     // AD type : name
+    memcpy(&advDataBuf[index], deviceName, mDeviceNameLength);         // AD value
```

Comment:
1. A name is not part of to-spec Matter advertising. If you add it, put it in the Scan Response instead. Spec text:

> In order to reduce 2.4 GHz spectrum congestion due to active BLE scanning, and to extend battery life in battery-powered devices, all critical data used for device discovery is contained in the Advertising Data rather than the Scan Response Data. This allows a BLE Commissioner to passively scan (i.e., not issue Scan Requests upon receiving scannable advertisements) and still be able to receive all information needed to commission a device.
> 
> Note that if additional vendor-specific information is to be conveyed and does not fit within the Advertising Data, it may be included in the Scan Response Data. See <<Manufacturer-specific data>> for details on including vendor-specific information.
> 

2. The bound checks above (line 460/461) do not validate that `deviceName` fits within the payload. This could overrun the advertisement data buffer.

Reply:
If this is existing functionality, it seems outside the scope of the PR but important to track. Issue may be appropriate, maybe add a TODO in the code.